authorization/invitation/service/invitation_service.go

```diff
@@ -162,3 +211,98 @@ func (s *invitationServiceImpl) Issue(ctx context.Context, issuingUserId uuid.UU
 
 	return nil
 }
+
+type invitationNotification struct {
+	invitation *invitationrepo.Invitation
+	roles      []string
+}
+
+// processTeamInviteNotifications sends an e-mail notification to a user.
+func (s *invitationServiceImpl) processTeamInviteNotifications(ctx context.Context, team *account.Identity, inviterName string,
+	notifications []invitationNotification, witURL string) error {
+	teamName := team.IdentityResource.Name
+
+	var spaceName string
+	var err error
+
+	if witURL != "" {
+		spaceName, err = lookupSpaceName(ctx, witURL, *team.IdentityResource.ParentResourceID)
+		if err != nil {
+			return err
+		}
+	}
+
+	var messages []notification.Message
+
+	for _, n := range notifications {
+		acceptURL := fmt.Sprintf("%s/api/invitations/accept?code=%s", s.config.GetAuthServiceURL(), n.invitation.AcceptCode.String())
+
+		messages = append(messages, notification.NewTeamInvitationEmail(n.invitation.Identity.UserID.UUID.String(),
+			teamName,
+			inviterName,
+			spaceName,
+			acceptURL))
+	}
+
+	s.Services().NotificationService().SendMessagesAsync(ctx, messages)
+
+	return nil
+}
+
+// processSpaceInviteNotifications sends an e-mail notification to a user.
+func (s *invitationServiceImpl) processSpaceInviteNotifications(ctx context.Context, space *resource.Resource,
+	inviterName string, notifications []invitationNotification, witURL string) error {
+
+	var spaceName string
+	var err error
+
+	if witURL != "" {
+		spaceName, err = lookupSpaceName(ctx, witURL, space.ResourceID)
+		if err != nil {
+			return err
+		}
+	}
+
+	var messages []notification.Message
+
+	for _, n := range notifications {
+		acceptURL := fmt.Sprintf("%s/api/invitations/accept?code=%s", s.config.GetAuthServiceURL(), n.invitation.AcceptCode.String())
+
+		messages = append(messages, notification.NewSpaceInvitationEmail(n.invitation.Identity.UserID.UUID.String(),
+			spaceName,
+			inviterName,
+			strings.Join(n.roles, ","),
+			acceptURL))
+	}
+
+	s.Services().NotificationService().SendMessagesAsync(ctx, messages)
+
+	return nil
+}
+
+// lookupSpaceName talks to the WIT service to retrieve a space record for the specified spaceID, then
+// returns the name of the space
+func lookupSpaceName(ctx context.Context, witURL string, spaceID string) (string, error) {
```

Comment:
This code should move to our WIT service (wit/wit.go). The WIT service needs to be refactored though. The same refactoring as we did for Notification service.
@sbryzak can you maybe create an issue for that? I will try to do such refactoring this week so you can consume it here.